api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1036,14 +1037,14 @@ private HttpResponse executeDelete(String bucketName, String objectName, Map<Str
    * @param data           HTTP request body data.
    */
   private HttpResponse executePost(String bucketName, String objectName, Map<String,String> headerMap,
-                                   Map<String,String> queryParamMap, Object data)
+                                   Map<String,String> queryParamMap, String contentType, Object data)
```

Comment:
This leads to a lot more involved, i would do a cleanup in a separate patch @balamurugana - because this patch at-least fixes the original problem of regression.